src/view/treeNodes/descriptionNode.ts

```diff
@@ -32,12 +38,63 @@ export class DescriptionNode extends TreeNode implements vscode.TreeItem {
 			arguments: [this],
 		};
 
-		this.contextValue = 'description';
+		this.registerSinceReviewChange();
+
 		this.tooltip = `Description of pull request #${pullRequestModel.number}`;
 		this.accessibilityInformation = { label: `Pull request page of pull request number ${pullRequestModel.number}`, role: 'button' };
 	}
 
 	getTreeItem(): vscode.TreeItem {
 		return this;
 	}
+
+	updateContextValue(): void {
+		this.contextValue = 'description' +
+			(this.pullRequestModel.hasChangesSinceLastReview ? ':changesSinceReview' : '') +
+			(this.pullRequestModel.showChangesSinceReview ? ':active' : ':inactive');
+	}
+
+	protected registerSinceReviewChange() {
+		this.childrenDisposables.push(
+			this.pullRequestModel.onDidChangeChangesSinceReview(async data => {
+				const { openFirst } = data;
+				this.updateContextValue();
+				this.refresh();
+				await this.reopenNewPrDiffs(this.pullRequestModel);
+				if (openFirst) {
+					PullRequestModel.openFirstDiff(this._folderReposManager, this.pullRequestModel);
+				}
+			})
+		);
+	}
+
+	public async reopenNewPrDiffs(pullRequest: PullRequestModel) {
+		const proms = vscode.window.tabGroups.all.map(tabGroup => {
+			return tabGroup.tabs.map(tab => {
+				Promise.resolve(this.reopenTab(tab, pullRequest));
+			});
+		}).flat();
+
+		await Promise.all(proms);
+	}
+
+	public async reopenTab(tab: vscode.Tab, pullRequest: PullRequestModel) {
+		if (tab.input instanceof vscode.TabInputTextDiff) {
+			if ((tab.input.original.scheme === Schemes.Pr) && (tab.input.modified.scheme === Schemes.Pr)) {
+				const changes = await (this.parent as PRNode).getFileChanges(true); // HERE THE 'UGLY' WORKAROUND
```

Comment:
Here is where we are looking for a solution as we've discussed. After refreshing we reopen the diffs but the parent node has not yet received the new changes.

Reply:
Can we move basically all of this, including the `onDidChangeChangesSinceReview` listener, into `PRNode`? That way, we can explicitly call and await `getFileChanges` before doing a `refresh`.

Reply:
Yes, this works a lot nicer now in the PRNode

Reply:
Looks good!